src/discovery.rs

```diff
@@ -0,0 +1,75 @@
+use crate::errors::KaboodleError;
+use crate::networking::create_broadcast_sockets;
+use crate::structs::{SwimBroadcast, SwimEnvelope};
+use bytes::Bytes;
+use if_addrs::Interface;
+use tokio::time::timeout;
+
+use std::{
+    net::SocketAddr,
+    time::{Duration, Instant},
+};
+use tokio::net::UdpSocket;
+
+/// How large a buffer to use when reading from our sockets; messages larger than this will be
+/// truncated.
+const INCOMING_BUFFER_SIZE: usize = 1024;
```

Comment:
Would it make sense to make this the same (10240) value as [this](https://github.com/serval/kaboodle/blob/4ef23a46277e6828e7b4413b592dd888f3549078/src/kaboodle.rs#L40)?

Reply:
It's intentionally smaller. The main `INCOMING_BUFFER_SIZE` needs to be large enough to account for a message with a list of known peers, which is variable and unbounded in size. This particular module only ever needs to be big enough to handle a `ProbeResponse`, and while that is theoretically variable and unbounded in size because it holds an `identity` payload, the expectation is that those are always reasonable small (likely < 100 bytes).